### PR TITLE
Update testplan with GCP/Azure auto-discovery

### PR DESCRIPTION
**Pull request by atburke**
This PR updates the testplan to include GCP and Azure VM auto-discovery.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/testplan.md [hunk 1]
```diff
@@ -1262,6 +1263,29 @@ TODO(lxea): replace links with actual docs once merged
...
+  - [ ] Only Azure VMs matching given Azure tags have the installer executed on them
+  - [ ] Only the IAM permissions mentioned in the discovery docs are required for operation
+  - [ ] Custom scripts specified in different matchers are executed
+  - [ ] New Azure VMs with matching Azure tags are discovered and added to the teleport cluster
+    - [ ] Large numbers of Azure VMs (51+) are all successfully added to the cluster
+  - [ ] Nodes that have been discovered do not have the install script run on the node multiple times
+
+## GCP Discovery
+
+[GCP Discovery docs](https://goteleport.com/docs/server-access/guides/gcp-discovery/)
```
Comment (codingllama) [inline .github/ISSUE_TEMPLATE/testplan.md:1279]: Link 404s, is it correct? In review, maybe?

**Review by codingllama — APPROVED** [review]
LGTM, although you might want someone who knows more about discovery in general to take a look.

**Review by r0mant — APPROVED** [review]
We should create an action item to automate Azure/GCP discovery testing similar to what @tigrato and @GavinFrazar are doing with AWS. I'll file one.

File: .github/ISSUE_TEMPLATE/testplan.md (hunk 1 above)
Comment (atburke) [reply .github/ISSUE_TEMPLATE/testplan.md:1279]: It's in review. #29840

**Pull request merged**